src/test/java/com/firebase/geofire/TestListener.java

```diff
@@ -10,12 +10,12 @@
 import java.util.concurrent.locks.Condition;
 import java.util.concurrent.locks.ReentrantLock;
 
-public class TestListener {
+abstract class TestListener {
```

Comment:
Why abstract with no abstract methods?

Reply:
It's used in both listeners. We could probably use delegation instead and get rid of inheritance at all.